packages/alchemy/src/client/factory.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  erc4337ClientActions,
+  type UserOperationRequest,
+} from "@alchemy/aa-core";
+import {
+  createClient,
+  http,
+  publicActions,
+  size,
+  slice,
+  type Address,
+  type Chain,
+  type Client,
+  type HttpTransport,
+  type HttpTransportConfig,
+  type Transport,
+} from "viem";
+import type { DebugTransaction, SimulateExecutionResponse } from "./interfaces";
+import type {
+  AlchemyEnhanced4337Client,
+  AlchemyEnhancedApiSchema,
+} from "./types";
+
+export const alchemyEnhancedApiActions = (client: Client) => {
+  const clientAdapter = client as Client<
+    Transport,
+    Chain,
+    undefined,
+    AlchemyEnhancedApiSchema
+  >;
+
+  return {
+    simulateUserOperationExecution(
+      request: UserOperationRequest
+    ): Promise<SimulateExecutionResponse | SimulateExecutionResponse[]> {
+      const { sender, initCode, callData } = request;
+      const entryPointAddress: Address =
+        "0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789";
+
+      const innerTransactionObject: DebugTransaction = {
+        from: entryPointAddress,
+        to: sender,
+        data: callData,
+      };
+
+      if (initCode === "0x") {
+        return clientAdapter.request({
+          method: "alchemy_simulateExecution",
+          params: [innerTransactionObject],
+        });
+      }
```

Comment:
https://docs.alchemy.com/reference/alchemy-simulateexecution

Does this api support BASE chain or not? Either this linked doc is outdated, or if not, we should add a check if the chain is Base chain until the api gets updated with Base support.

Reply:
good point, lemme check with Malak